src-web/components/kappnav/common/ResourceTable.js

```diff
@@ -40,6 +40,39 @@ const {
 
 const translateWithId = (locale, id) => msgs.get(id)
 
+// returing the field if its of "string" type
+export const ifTheFieldIsNotLink = (row, field) => {
+	if (row && field && typeof field === 'string') {
+		return field
+	}
+}
+
+// function to get all the cell values of the DataTable
+export const getSearchableCellList = (row, headers) => {
+	var cellValues = []
+	headers.forEach(header => {
+		// for "status" field
+		if (header.type === 'status') {
+			if (row && row[header.key] && row[header.key].props) {
+				// Since "status" can't be a string or a Link, hence finding its value under the "props" key
+				cellValues.push(row[header.key].props.children[1].props.children)
+			}
+		} else {
+			// for feilds other than "menuAction", "title", "description", "section_data" and "section_map" as these are not searchable 
+			if (header.key !== "menuAction" && header.key !== "title" && header.key !== "description" && header.key !== "section_data" && header.key !== "section_map") {
```

Comment:
We should not need to refer to the contents of header.key.  Instead if a header is not searchable, then give it a type of not searchable.